discord/utils/public.py

```diff
@@ -0,0 +1,536 @@
+from __future__ import annotations
+
+import asyncio
+import re
+import datetime
+from enum import Enum, auto
+import itertools
+from collections.abc import Awaitable, Callable, Iterable
+from typing import TYPE_CHECKING, Any, Literal, Iterable, TypeVar
```

Comment:
```suggestion
from typing import TYPE_CHECKING, Any, Literal, Iterable, TypeVar, Match
```
used for the typing of match